src/txn/txn_rollback_to_stable.c

```diff
@@ -1438,9 +1438,11 @@ __rollback_to_stable_hs_final_pass(WT_SESSION_IMPL *session, wt_timestamp_t roll
 
     /*
      * The rollback operation should be performed on the history store file when the checkpoint
-     * durable start/stop timestamp is greater than the rollback timestamp.
+     * durable start/stop timestamp is greater than the rollback timestamp. But skip if there are
+     * timestamped updates with no stable timestamp.
      */
-    if (max_durable_ts > rollback_timestamp) {
+    if (max_durable_ts > rollback_timestamp &&
+      (rollback_timestamp != WT_TS_NONE || max_durable_ts == WT_TS_NONE)) {
```

Comment:
I don't clearly understand the reason behind `max_durable_ts == WT_TS_NONE` change? Can you explain to me the reason behind that change?

Reply:
It is the same as the corresponding check in __rollback_to_stable_btree_apply (skip if there's no stable timestamp, but only for tables that have timestamps). It probably doesn't matter since if there are no timestamps there should be no history, but it seemed better to make the logic the same in both places.

Reply:
Should maybe stuff that information in the comment I guess.

Reply:
On second thought because the rest of the expression is also testing max_durable_ts it's kind of silly to have that check since it's impossible. Will update.

Reply:
Done.

Reply:
Thanks.